### PR TITLE
LIB-226: Move from TravisCI to GitHub actions

### PR DESCRIPTION
**Pull request by kennylindley**
Removed the TravisCI configuration and setup config for github actions. Updated the badge in README.md to point to the GitHub actions badge.

NOTES:
* For this to be successful, a new Github Secret called CC_TEST_REPORTER_ID needs to be populated with our test reporter ID. I have already done this.
* Had to manually install `sprockets-rails` for the latest appraisal
Also, I presume that this action will not run until it is merged into master.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: tests
+on: [pull_request]
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [2.5.7, 2.6.5, 2.7.0]
```
Comment (bmedenwald) [inline .github/workflows/tests.yml:9]: We should make some updates here, like I did with Puma Pool Usage. At minimum, there should be something in the Ruby 3.x line. I think:

`ruby: [2.7.6, 3.0.4, 3.1.2]`

If you make the minimum 2.7.6, you can remove `gem install bundler`, `gem update`, and `bundle install`, which would happen automatically.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    env:
+      CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+      RAILS_ENV: test
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
```
Comment (bmedenwald) [inline .github/workflows/tests.yml:17]: Should add:

`bundler-cache: true # runs 'bundle install' and caches installed gems automatically`

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - uses: supercharge/redis-github-action@1.4.0
+      - run: gem install bundler -v "< 2"
+      - run: gem update --system 3.2.3
+      - run: bundle install
+      - name: Install Code Climate
+        run: |
+          curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+          chmod +x ./cc-test-reporter
+          ./cc-test-reporter before-build
```
Comment (bmedenwald) [inline .github/workflows/tests.yml:26]: Add a blank line below this to separate all the steps.

File: Appraisals [hunk 4]
```diff
@@ -11,4 +11,5 @@ end
 appraise "latest" do
   gem "rails"
   gem "sidekiq"
+  gem "sprockets-rails"
```
Comment (bmedenwald) [inline Appraisals:14]: I doubt you need this. If you do, please include the errors that happen if you don't so I can look into it further.

**Review by bmedenwald — CHANGES_REQUESTED** [review]

File: Appraisals (hunk 4 above)
Comment (kennylindley) [reply Appraisals:14]:
```
Failure/Error: require "sprockets/railtie"

LoadError:
  cannot load such file -- sprockets/railtie
# ./spec/dummy/config/application.rb:5:in `require'
# ./spec/dummy/config/application.rb:5:in `<top (required)>'
# ./spec/dummy/config/environment.rb:2:in `require_relative'
# ./spec/dummy/config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:4:in `require'
# ./spec/rails_helper.rb:4:in `<top (required)>'
# ./spec/simple_scheduler/at_spec.rb:1:in `require'
# ./spec/simple_scheduler/at_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/simple_scheduler/future_job_spec.rb. - Did you mean?
                    rspec ./spec/simple_scheduler/scheduler_job_spec.rb
                    rspec ./spec/simple_scheduler/at_spec.rb
                    rspec ./spec/simple_scheduler/task_spec.rb
```

**Pull request closed without merging**